src/implicit-function.hh

```diff
@@ -0,0 +1,320 @@
+    }
+
+    template <typename LgT> void JacobianVisitor::operator () (const LgT&)
+    {
+      for (size_type i=0; i<outJacobian_.nbRows (); ++i) {
+        outJacobian_.lview (result_) (i, i) = 1;
+      }
+      inJacobian_.lview (result_) = -Jf_;
+    }
+      
+
+    HPP_PREDEF_CLASS (ImplicitFunction);
+    typedef boost::shared_ptr <ImplicitFunction> ImplicitFunctionPtr_t;
+
+    /// Function of the form f (q) = q2 - g (q1)
+    ///
+    /// where
+    ///  \li q2 is a vector composed of a subset of configuration variables of
+    ///      q,
+    ///  \li q1 is the vector composed of the other configuration variables of
+    ///      q,
+    ///  g is a differentiable function with values in  a Lie group.
+    class ImplicitFunction : public DifferentiableFunction
+    {
+    public:
+      static ImplicitFunctionPtr_t create
+      (const DevicePtr_t& robot, const DifferentiableFunctionPtr_t& function,
+       const segments_t& inputConf, const segments_t& inputVelocity,
+       const segments_t& outputConf, const segments_t& outputVelocity)
+      {
+	ImplicitFunction* ptr = new ImplicitFunction
+	  (robot, function, inputConf, inputVelocity, outputConf,
+           outputVelocity);
+	return ImplicitFunctionPtr_t (ptr);
+      }
+
+      /// Get function that maps input variables to output variables
+      const DifferentiableFunctionPtr_t& inputToOutput () const
+      {
+        return inputToOutput_;
+      }
+
+    protected:
+      ImplicitFunction (const DevicePtr_t& robot,
+			const DifferentiableFunctionPtr_t& function,
+			const segments_t& inputConf,
+			const segments_t& inputVelocity,
+                        const segments_t& outputConf,
+			const segments_t& outputVelocity)
+	: DifferentiableFunction (robot->configSize (), robot->numberDof (),
+				  LiegroupSpace::Rn
+                                  (function->outputSpace ()->nv ())),
+	  robot_ (robot), inputToOutput_ (function),
+          inputConfIntervals_ (inputConf),
+	  inputDerivIntervals_ (inputVelocity),
+          outputConfIntervals_ (outputConf),
+	  outputDerivIntervals_ (outputVelocity), outJacobian_ (),
+          inJacobian_ (), f_qIn_ (function->outputSpace ()),
+          qOut_ (function->outputSpace ()), result_ (outputSpace ())
+      {
+	// Check input consistency
+	// Each configuration variable is either input or output
+	assert (function->inputSize () + function->outputSize () <=
+		robot->configSize ());
+	// Each velocity variable is either input or output
+	assert (function->inputDerivativeSize () +
+		function->outputDerivativeSize () <= robot->numberDof ());
+	qIn_.resize (function->inputSize ());
+	Jf_.resize (function->outputDerivativeSize (),
+                    function->inputDerivativeSize ());
+	size_type size = 0;
+	// Sum of configuration output interval sizes equal function output size
+	for (segments_t::const_iterator it = outputConf.begin ();
+	     it != outputConf.end (); ++it) {
+	  size += it->second;
+	}
```

Comment:
This could probably be replaced by `BlockIndex::cardinal (outputConf);`

Reply:
I will modify